track ids independent of groups in bed_cluster #171

kriemo opened the pull request:
Tracking the cluster `.id` was easier to implement on the Rcpp side, due to grouping issues. I also added a test and a note in the documentation that `bed_cluster()` requires sorted input.

jayhesselberth commented on R/bed_cluster.r:
```diff
@@ -1,20 +1,21 @@
-#' book-ended intervals will be clustered.
-#' 
-#' @param x tbl of intervals
+#' book-ended intervals will be clustered. Note that input tbl is assumed to be sorted
```
I would prefer to just sort this in the bed_cluster function itself, rather than requiring  explicit sort operation (one less thing to remember to do).

jayhesselberth commented on R/bed_cluster.r:
```diff
@@ -35,20 +36,19 @@
 
   res <- group_by(x, chrom, add = TRUE)
-  
+
```
add `res <- arrange(res, chrom, start)` here

kriemo commented:
I added the sort and fixed the docs. [Travis](https://www.traviscistatus.com/) is moving slow due to the AWS outage, so we will probably have to wait until tomorrow to check the builds.
